### ISSUE TITLE
Consider disallowing a submission of dataset URL that is a path of the file system

### ISSUE THREAD

**Issue by candleindark** [datalad/datalad-registry#248]
Currently, as it was allowed before the API 2.0 change, a user is allowed to submit a dataset URL that is actually a file system path.

@yarikoptic, I think this is a potential security issue. We should not allow a user to submit a URL that is a file system path.

Currently, a URL submitted by a user is validated by the following Pydantic model.

https://github.com/datalad/datalad-registry/blob/bdf42bd2dd7f511a9682b77b1354776f88f58dc2/datalad_registry/blueprints/api/dataset_urls/models.py#L177-188

**Comment by yarikoptic** [comment]
yes, let's disallow local paths as URLs, and `file://`, `http://localhost` etc urls. May be eventually  we should indeed run this somewhere in the cloud to really avoid possibly operating behind some "closed curtains" ;-)  ATM we are cloning only the ones found on github so we should be ok.
